### ISSUE TITLE
YOLOv5 Hyperparameter Evolution Results Not Reproducible

### ISSUE THREAD

**Issue by 25benjaminli** [ultralytics/yolov5#11032]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I cannot reproduce the results of hyperparameter evolution when I train a model on the derived optimal hyperparameters. 

For context, I am running the following command for hyperparameter evolution.
```
python train.py --batch 32 --epochs 10 --data ./final_ds/data.yaml --cache --weights yolov5s.pt --img 416 --evolve 100
```

I am running the following command to train my model based on the hyperparameters generated from the above command. 

`!python train.py --batch 32 --epochs 40 --data ./final_ds/data.yaml --cache --weights yolov5s.pt --img 416 --hyp runs/evolve/exp/hyp_evolve.yaml`

Apparently, the mAP and mAP@[.5:.95] of the optimal hyperparameter selection from the evolution algorithm are 0.72188 and 0.49567, respectively, but on the tenth epoch of the actual training, they were only 0.53078 and 0.29805.

I also noticed that the loss decreased significantly faster during the evolution algorithm than regular training - box loss reached 0.019 at around epoch 10 vs. 0.35 at epoch 10 for regular training. 

### Environment

- YOLO: YOLOv5 - v7.0
- Google colab
- Python 3.7

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!
